gravpm.c

```diff
@@ -265,6 +294,86 @@ static double sinc_unnormed(double x) {
     }
 }
 
+/* Compute neutrino power spectrum.
+ * This should happen after the CFT is computed,
+ * and after powerspectrum_compute() has been called,
+ * but before potential_transfer is called.*/
+static void compute_neutrino_power() {
+    if(!All.MassiveNuLinRespOn)
+        return;
+    /*Note the power spectrum is now in Mpc units*/
+    powerspectrum_sum(&PowerSpectrum, All.BoxSize*All.UnitLength_in_cm);
+    int i;
+    /*Get delta_cdm_curr , which is P(k)^1/2, and skip bins with zero modes:*/
+    int nk_nonzero = 0;
+    for(i=0;i<PowerSpectrum.size;i++){
+        if (PowerSpectrum.Nmodes[i] == 0)
+            continue;
+        PowerSpectrum.Pnuratio[nk_nonzero] = sqrt(PowerSpectrum.P[i]);
+        PowerSpectrum.k[nk_nonzero] = PowerSpectrum.k[i];
+        nk_nonzero++;
+    }
+    double Pnu[nk_nonzero];
+    memset(Pnu,0, nk_nonzero*sizeof(double));
+    /*This sets up P_nu_curr.*/
+    /*This is done on the first timestep: we need nk_nonzero for it to work.*/
+    if(!delta_tot_table.delta_tot_init_done) {
+        _transfer_init_table transfer_init;
+        if(ThisTask == 0) {
+            allocate_transfer_init_table(&transfer_init, All.BoxSize, 3.085678e24, All.CAMBInputSpectrum_UnitLength_in_cm, All.CAMBTransferFunction);
+        }
+        /*Broadcast the transfer size*/
+        MPI_Bcast(&(transfer_init.NPowerTable), 1,MPI_INT,0,MPI_COMM_WORLD);
+        /*Allocate the memory unless we are on task 0, in which case it is already allocated*/
+        if(ThisTask != 0)
+          transfer_init.logk = (double *) mymalloc("Transfer_functions", 2*transfer_init.NPowerTable* sizeof(double));
+        transfer_init.T_nu=transfer_init.logk+transfer_init.NPowerTable;
+        /*Broadcast the transfer table*/
+        MPI_Bcast(transfer_init.logk,2*transfer_init.NPowerTable,MPI_DOUBLE,0,MPI_COMM_WORLD);
+        /*Initialise delta_tot*/
+        delta_tot_init(&delta_tot_table, nk_nonzero, PowerSpectrum.k, PowerSpectrum.P, &transfer_init, All.Time);
```

Comment:
It may be useful to put all k-space-neutrino functions under the same prefix. not now though.